app/src/main/java/org/fossasia/susi/ai/settings/ChatSettingsFragment.kt

```diff
@@ -77,19 +81,40 @@ class ChatSettingsFragment : PreferenceFragmentCompat() {
 
         theme?.setOnPreferenceChangeListener({ preference, newValue ->
             preference.summary = newValue.toString()
-            settingsPresenter?.setTheme(newValue.toString())
+            settingsPresenter.setTheme(newValue.toString())
+            settingsPresenter.sendSetting("theme", newValue.toString())
             activity.recreate()
             true
         })
 
-        micSettings?.isEnabled = settingsPresenter?.enableMic() as Boolean
+        micSettings?.isEnabled = settingsPresenter.enablemic()
+
+        hotwordSettings?.isEnabled = settingsPresenter.enableHotword()
 
-        hotwordSettings?.isEnabled = settingsPresenter?.enableHotword() as Boolean
+        micSettings?.setOnPreferenceClickListener {
+            settingsPresenter.sendSetting("mic_input", (PrefManager.getBoolean(Constant.MIC_INPUT, false)).toString())
```

Comment:
Which constant? on server we are saving setting value in string format. So I am converting the value of mic setting to string.